vhdbuilder/packer/init-variables.sh

```diff
@@ -340,72 +340,33 @@ fi
-	imported_windows_image_name=""
-	source $CDIR/windows/windows-image.env
-
-	echo "Set the base image sku and version from windows-image.env"
-	case "${WINDOWS_SKU}" in
-	"2019")
-		WINDOWS_IMAGE_SKU=$WINDOWS_2019_BASE_IMAGE_SKU
-		WINDOWS_IMAGE_VERSION=$WINDOWS_2019_BASE_IMAGE_VERSION
-		imported_windows_image_name="windows-2019-imported-${CREATE_TIME}-${RANDOM}"
-
-		echo "Set OS disk size"
-		if [ -n "${WINDOWS_2019_OS_DISK_SIZE_GB}" ]; then
-			echo "Setting os_disk_size_gb to the value in windows-image.env for 2019 Docker: ${WINDOWS_2019_OS_DISK_SIZE_GB}"
-			os_disk_size_gb=${WINDOWS_2019_OS_DISK_SIZE_GB}
-		fi
-		;;
-	"2019-containerd")
-		WINDOWS_IMAGE_SKU=$WINDOWS_2019_BASE_IMAGE_SKU
-		WINDOWS_IMAGE_VERSION=$WINDOWS_2019_BASE_IMAGE_VERSION
-		imported_windows_image_name="windows-2019-containerd-imported-${CREATE_TIME}-${RANDOM}"
-
-		echo "Set OS disk size"
-		if [ -n "${WINDOWS_2019_CONTAINERD_OS_DISK_SIZE_GB}" ]; then
-			echo "Setting os_disk_size_gb to the value in windows-image.env for 2019 Containerd: ${WINDOWS_2019_CONTAINERD_OS_DISK_SIZE_GB}"
-			os_disk_size_gb=${WINDOWS_2019_CONTAINERD_OS_DISK_SIZE_GB}
-		fi
-		;;
-	"2022-containerd" | "2022-containerd-gen2")
-		WINDOWS_IMAGE_SKU=$WINDOWS_2022_BASE_IMAGE_SKU
-		WINDOWS_IMAGE_VERSION=$WINDOWS_2022_BASE_IMAGE_VERSION
-		imported_windows_image_name="windows-2022-containerd-imported-${CREATE_TIME}-${RANDOM}"
-
-		echo "Set OS disk size"
-		if [ -n "${WINDOWS_2022_CONTAINERD_OS_DISK_SIZE_GB}" ]; then
-			echo "Setting os_disk_size_gb to the value in windows-image.env for 2022 Containerd: ${WINDOWS_2022_CONTAINERD_OS_DISK_SIZE_GB}"
-			os_disk_size_gb=${WINDOWS_2022_CONTAINERD_OS_DISK_SIZE_GB}
-		fi
-		# Default: read from the official MCR image
-		if [[ $HYPERV_GENERATION == "V2" ]]; then
-			WINDOWS_IMAGE_SKU=$WINDOWS_2022_GEN2_BASE_IMAGE_SKU
-			WINDOWS_IMAGE_VERSION=$WINDOWS_2022_GEN2_BASE_IMAGE_VERSION
-		fi
-		;;
-	"23H2" | "23H2-gen2")
-		WINDOWS_IMAGE_SKU=$WINDOWS_23H2_BASE_IMAGE_SKU
-		WINDOWS_IMAGE_VERSION=$WINDOWS_23H2_BASE_IMAGE_VERSION
-		imported_windows_image_name="windows-23H2-imported-${CREATE_TIME}-${RANDOM}"
-
-		echo "Set OS disk size"
-		if [ -n "${WINDOWS_23H2_OS_DISK_SIZE_GB}" ]; then
-			echo "Setting os_disk_size_gb to the value in windows-image.env for 23H2: ${WINDOWS_23H2_OS_DISK_SIZE_GB}"
-			os_disk_size_gb=${WINDOWS_23H2_OS_DISK_SIZE_GB}
-		fi
-		# Default: read from the official MCR image
-		if [[ $HYPERV_GENERATION == "V2" ]]; then
-			WINDOWS_IMAGE_SKU=$WINDOWS_23H2_GEN2_BASE_IMAGE_SKU
-			WINDOWS_IMAGE_VERSION=$WINDOWS_23H2_GEN2_BASE_IMAGE_VERSION
-		fi
-		;;
-	*)
-		echo "unsupported windows sku: ${WINDOWS_SKU}"
-		exit 1
-		;;
-	esac
-
-	# Create the sig image from the official images defined in windows-image.env by default
+
+	echo "Set the base image sku and version from windows-settings.json"
+
+	WINDOWS_IMAGE_SKU=`cat $CDIR/windows/windows_settings.json | jq -r ".WindowsBaseVersions.\"${WINDOWS_SKU}\".base_image_sku"`
```

Comment:
Gentle reminder: when I did jq in the Linux VHD case, there were some cases that the scripts crashed because some json files were not found in older VHDs. This did cause IcM at that time. But this is because in Linux node, we have to match CSE scripts and VHDs. Not sure if it applies to Windows. Just an FYI.